automation/taskcluster/decision_task.py

```diff
@@ -185,62 +190,108 @@ def gradle_module_task(libs_tasks, module_info, is_release):
     return task.create()
 
-def build_gradle_modules_tasks(is_release):
-    libs_tasks = libs_for("android", "desktop_linux", "desktop_macos", "desktop_win32_x86_64")
+def build_gradle_modules_tasks(deploy_environment):
+    libs_tasks = libs_for(deploy_environment, "android", "desktop_linux", "desktop_macos", "desktop_win32_x86_64")
     module_build_tasks = {}
     for module_info in module_definitions():
-        module_build_tasks[module_info['name']] = gradle_module_task(libs_tasks, module_info, is_release)
+        module_build_tasks[module_info['name']] = gradle_module_task(libs_tasks, module_info, deploy_environment)
     return module_build_tasks
 
 def android_multiarch():
     ktlint_detekt()
-    build_gradle_modules_tasks(False)
+    build_gradle_modules_tasks(DeployEnvironment.NONE)
 
-def android_multiarch_release():
-    module_build_tasks = build_gradle_modules_tasks(True)
+def android_multiarch_release(is_staging):
+    module_build_tasks = build_gradle_modules_tasks(DeployEnvironment.STAGING_RELEASE if is_staging else DeployEnvironment.RELEASE)
 
     version = appservices_version()
-    worker_type = os.environ['BEETMOVER_WORKER_TYPE']
     bucket_name = os.environ['BEETMOVER_BUCKET']
     bucket_public_url = os.environ['BEETMOVER_BUCKET_PUBLIC_URL']
 
     for module_info in module_definitions():
         module = module_info['name']
         build_task = module_build_tasks[module]
-        for artifact in module_info['artifacts']:
-            artifact_name = artifact['name']
-            artifact_path = artifact['path']
-            (
-                BeetmoverTask("Publish Android module: {} via beetmover".format(artifact_name))
-                .with_description("Publish release module {} to {}".format(artifact_name, bucket_public_url))
-                .with_worker_type(worker_type)
-                # We want to make sure ALL builds succeeded before doing a release.
-                .with_dependencies(*module_build_tasks.values())
-                .with_upstream_artifact({
-                    "paths": [artifact_path],
-                    "taskId": build_task,
-                    "taskType": "build",
-                    "zipExtract": True,
-                })
-                .with_app_name("appservices")
-                .with_artifact_id(artifact_name)
-                .with_app_version(version)
-                .with_scopes(
-                    "project:mozilla:application-services:releng:beetmover:bucket:{}".format(bucket_name),
-                    "project:mozilla:application-services:releng:beetmover:action:push-to-maven"
-                )
-                .with_routes("notify.email.a-s-ci-failures@mozilla.com.on-failed")
-                .create()
+        sign_task = (
+            SignTask("Sign Android module: {}".format(module))
+            .with_description("Signs module")
+            .with_worker_type("appsv-signing-dep-v1" if is_staging else "appsv-signing-v1")
```

Comment:
Nit: is it worth being consistent here to define similar env vars for SIGNING_* as well so that we have SIGNING_WORKER_TYPE instead?

Reply:
TBH, I'd rather move in the other direction: it's easier to add re-use logic to python (the decision task) than to JSON-e. At the same time, I didn't move to make it consistent (everything being set in the decision task based on `--staging`) because `BEETMOVER_*` are environment variables, and I didn't want to be hit by unexpected breakage because some layer deep down depended on 'em.

Reply:
Okay, that seems reasonable. FTR, I prefer the python X if `is_staging` else Y rather than env vars 😊

Reply:
Big :+1:. My plan is to perhaps move the `BEETMOVER_*` variables into python in a follow-up some time in the future

Reply:
Crisp!